src/index.js

```diff
@@ -1,52 +1,99 @@
   {
     const toDoTaskStatus = document.createElement('input');
     toDoTaskStatus.type = 'checkbox';
     toDoTaskStatus.checked = task.completed;
+    toDoTaskStatus.addEventListener('change', () => {
+      if (!toDoTaskStatus.checked) toDoTaskStatus.nextSibling.style.textDecoration = 'none';
+      else toDoTaskStatus.nextSibling.style.textDecoration = 'line-through';
+      updateListItem(toDoTaskStatus.checked, task.index);
+    });
     toDoTaskStatus.classList.add('task-status');
     toDoTaskItem.appendChild(toDoTaskStatus);
   }
   {
-    const toDoTaskDescription = document.createElement('p');
-    toDoTaskDescription.textContent = task.description;
+    const toDoTaskDescription = document.createElement('textarea');
+    toDoTaskDescription.value = task.description;
     toDoTaskDescription.classList.add('task-description');
+
+    if (toDoTaskItem.firstChild.checked) toDoTaskDescription.style.textDecoration = 'line-through';
+
+    toDoTaskDescription.addEventListener('keyup', (event) => {
+      toDoTask.updateDescription(event.target.value, task.index);
+    });
+
+    toDoTaskDescription.addEventListener('focus', (event) => {
+      focusedItem = true;
+      event.target.nextSibling.src = deleteIcon;
+      event.target.nextSibling.alt = 'delete';
+      event.target.nextSibling.className = 'delete-icon';
+      event.target.style.backgroundColor = '#fffeca';
+      event.target.parentElement.style.backgroundColor = '#fffeca';
+    });
+
+    toDoTaskDescription.addEventListener('blur', (event) => {
+      focusedItem = false;
+      event.target.nextSibling.src = dragDropIcon;
+      event.target.nextSibling.alt = 'drag drop';
+      event.target.nextSibling.className = 'task-drag-icon';
+      event.target.style.backgroundColor = '';
+      event.target.parentElement.style.backgroundColor = '';
+    });
+
     toDoTaskItem.appendChild(toDoTaskDescription);
   }
   {
     const toDoTaskplaceicon = document.createElement('img');
     toDoTaskplaceicon.src = dragDropIcon;
-    toDoTaskplaceicon.checked = task.completed;
-    toDoTaskplaceicon.classList.add('drop-drag-icon');
+    toDoTaskplaceicon.alt = 'drag drop';
+    // toDoTaskplaceicon.checked = task.completed;
+    toDoTaskplaceicon.className = 'drop-drag-icon';
+    toDoTaskplaceicon.addEventListener('mousedown', () => {
+      if (focusedItem) {
+        toDoTaskItem.remove();
+        toDoTask.remove(task.index);
+      }
+    });
     toDoTaskItem.appendChild(toDoTaskplaceicon);
   }
   return toDoTaskItem;
 }
 
 function updateTaskListToHTML() {
-  toDoTask.forEach((task) => document.getElementById('todo-list').appendChild(getTaskList(task)));
+  toDoTask.taskList.forEach((task) => document.getElementById('todo-list').appendChild(getTaskList(task)));
+}
+function addTask(description) {
+  if (description) { document.getElementById('todo-list').appendChild(getTaskList(toDoTask.add(description))); }
 }
 
+const addTaskElement = document.getElementById('list-add');
```

Comment:
- [ ] Nice working on your add task function, kindly check this block of code it doesn't seem to work when adding a task and getting tasks, I get this error in the console on page load.

![error](https://user-images.githubusercontent.com/48371156/184523337-d27940b8-6017-4110-b570-306ed25377d0.png)


- [ ] Kindly remove all commented codes.